### PR TITLE
Completed lesson 4-1

### PR DESCRIPTION
**Pull request by camerongineer**

### CODE REVIEW

File: src/components/TodoList.js [hunk 1]
```diff
@@ -10,4 +11,9 @@ const TodoList = ({ todoList, onRemoveTodo }) => (
         />)}
     </ul>);
 
+TodoList.propTypes = {
+    todoList: PropTypes.array.isRequired,
```
Comment (mattcarlotta) [inline src/components/TodoList.js]: Stretch Goal:
What is `todoList` an array of? An array of strings, numbers, or an object (shape) that has structure!? When it comes to arrays and objects, the more specific you are in their structure, the better type checking you'll have during run-time!

<details>
<summary>Click here for answer</summary>

```js
TodoList.propTypes = {
  todoList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }),
  ),
  onRemoveTodo: PropTypes.func.isRequired,
}
```

</details>

File: src/components/TodoListItem.js [hunk 2]
```diff
@@ -15,4 +16,9 @@ const TodoListItem = ({ item, onRemoveClicked }) =>
         </div>
     </li>;
 
+TodoListItem.propTypes = {
+    item: PropTypes.object.isRequired,
```
Comment (mattcarlotta) [inline src/components/TodoListItem.js]: Stretch Goal:
What is `item` an object of (what's its shape)?

**Review by mattcarlotta — APPROVED** [review]
You nailed this assignment, great work Cameron! 👍 

I added some optional stretch goals if you want to take your code to the next level.

File: src/components/TodoListItem.js (hunk 2 above)
Comment (camerongineer) [reply src/components/TodoListItem.js]: item: PropTypes.shape({
        id: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
    })

File: src/components/TodoList.js (hunk 1 above)
Comment (camerongineer) [reply src/components/TodoList.js]: todoList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }),
  ),

File: src/components/TodoListItem.js (hunk 2 above)
Comment (camerongineer) [reply src/components/TodoListItem.js]: Thanks for the tip! I went ahead and added the specific types.

**Pull request merged**